File: src/app/src/main/java/com/onscreensync/tvapp/CurrentDateTimeActivity.java
package com.onscreensync.tvapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.text.Html;
import android.widget.TextView;

import com.onscreensync.tvapp.utils.ObjectExtensions;
import com.onscreensync.tvapp.utils.UiHelper;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;

public class CurrentDateTimeActivity extends AppCompatActivity {

    private TextView dateTimeTextView;
    private String dateTimeFormat;
    private String textColor;
    private String textFont;
    private Handler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_date_time);
        dateTimeTextView = findViewById(R.id.datetime_activity_text_view);

        // Retrieve the Intent that started this activity
        Intent intent = getIntent();
        dateTimeFormat = intent.getStringExtra("dateTimeFormat");
        textColor = intent.getStringExtra("textColor");
        textFont = intent.getStringExtra("textFont");

        UiHelper.setTextViewFont(dateTimeTextView, textFont);
        UiHelper.setTextViewColor(dateTimeTextView, textColor);

        if(ObjectExtensions.isNullOrEmpty(dateTimeFormat)) dateTimeFormat = "EEE, d MMM yyyy, HH:mm:ss";

        handler = new Handler();
        final Runnable r = new Runnable() {
            public void run() {
                displayCurrentDateTime();
                handler.postDelayed(this, 1000);
            }
        };
        handler.postDelayed(r, 1000);

    }

    @Override
    public void onBackPressed() {
        finishAffinity();
        finish();
    }

    private void displayCurrentDateTime()
    {
        DateFormat df = new SimpleDateFormat(dateTimeFormat);
        String date = df.format(Calendar.getInstance().getTime());
        dateTimeTextView.setText(Html.fromHtml(date));
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/CurrentDateTimeActivity.kt
package com.onscreensync.tvapp

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.text.Html
import androidx.appcompat.app.AppCompatActivity
import com.onscreensync.tvapp.databinding.ActivityDateTimeBinding
import com.onscreensync.tvapp.signalR.SignalRManager
import com.onscreensync.tvapp.utils.UiHelper
import dagger.hilt.android.AndroidEntryPoint
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale
import javax.inject.Inject

@AndroidEntryPoint
class CurrentDateTimeActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDateTimeBinding
    private var dateTimeFormat: String = "EEE, d MMM yyyy, HH:mm:ss"
    private val handler = Handler(Looper.getMainLooper())
    
    @Inject
    lateinit var signalRManager: SignalRManager

    private val updateTimeRunnable = object : Runnable {
        override fun run() {
            displayCurrentDateTime()
            handler.postDelayed(this, 1000)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDateTimeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        intent.getStringExtra("dateTimeFormat")?.let {
            if (it.isNotEmpty()) dateTimeFormat = it
        }
        
        val textColor = intent.getStringExtra("textColor")
        val textFont = intent.getStringExtra("textFont")

        textFont?.let {
            UiHelper.setTextViewFont(binding.datetimeActivityTextView, it)
        }

        textColor?.let {
            UiHelper.setTextViewColor(binding.datetimeActivityTextView, it)
        }

        handler.post(updateTimeRunnable)
    }

    override fun onDestroy() {
        super.onDestroy()
        handler.removeCallbacks(updateTimeRunnable)
    }

    override fun onBackPressed() {
        signalRManager.disconnect()

        handler.postDelayed({
            finishAffinity()
        }, 2000)
    }

    private fun displayCurrentDateTime() {
        try {
            val sdf = SimpleDateFormat(dateTimeFormat, Locale.getDefault())
            val date = sdf.format(Calendar.getInstance().time)
            binding.datetimeActivityTextView.text = Html.fromHtml(date)
        } catch (e: Exception) {
            binding.datetimeActivityTextView.text = "Invalid Format"
        }
    }
}
